src/code/audio/code_8019AF00.c

```diff
@@ -2047,80 +3060,388 @@ void AudioOcarina_TerminaWallGenerateNotes(void) {
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D8B4.s")
+        sPrevOcarinaPitch = sCurOcarinaPitch;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D8E4.s")
+        if (sOcarinaDisableTimer != 0) {
+            sOcarinaDisableTimer--;
+            if (sOcarinaDisableTimer == 0) {
+                AudioOcarina_SetInstrument(OCARINA_INSTRUMENT_OFF);
+            }
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019DF28.s")
+    AudioOcarina_UpdatePlayingStaff();
+    AudioOcarina_UpdatePlaybackStaff();
+    AudioOcarina_UpdateRecordingStaff();
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019DF64.s")
+void AudioOcarina_PlayLongScarecrowAfterCredits(void) {
+    static u8 sScarecrowAfterCreditsState = 0;
+    static u8 sScarecrowAfterCreditsIntrumentId = OCARINA_INSTRUMENT_DEFAULT;
+    static u16 sScarecrowAfterCreditsTimer = 1200;
+
+    switch (sScarecrowAfterCreditsState) {
+        case 0:
+            if (sScarecrowAfterCreditsTimer-- == 0) {
+                // OoT originally had 7 Ocarina instruments
+                // MM has more so this does not go all the way to the max
+                if (sScarecrowAfterCreditsIntrumentId < 7) {
+                    // set next ocarina instrument and restart
+                    sScarecrowAfterCreditsState++;
+                } else {
+                    // finished
+                    sScarecrowAfterCreditsState = 3;
+                    AudioOcarina_SetInstrument(OCARINA_INSTRUMENT_OFF);
+                }
+                sScarecrowAfterCreditsTimer = 1200;
+            }
+            break;
+
+        case 1:
+            Audio_SetSfxBanksMute(0);
+            AudioOcarina_SetInstrument(sScarecrowAfterCreditsIntrumentId);
+            AudioOcarina_SetPlaybackSong(OCARINA_SONG_SCARECROW_LONG + 1, 1);
+            sScarecrowAfterCreditsIntrumentId++;
+            sScarecrowAfterCreditsState++;
+            break;
+
+        case 2:
+            if (AudioOcarina_GetPlaybackStaff()->state == 0) {
+                sScarecrowAfterCreditsState = 0;
+            }
+            break;
+    }
+}
+
+u8 sRequestCustomSequence = false;
+
+void AudioOcarina_SetCustomSequence(void) {
+    // Never passes true as sRequestCustomSequence is never set true
+    if (sRequestCustomSequence && gAudioContext.seqPlayers[SEQ_PLAYER_FANFARE].enabled &&
+        ((u8)gAudioContext.seqPlayers[SEQ_PLAYER_FANFARE].soundScriptIO[0] == 0xFF)) {
+        gAudioContext.seqPlayers[SEQ_PLAYER_FANFARE].seqData = sCustomSequenceScript;
+        sRequestCustomSequence = false;
+    }
+}
+
+// Called by unused function
+void AudioOcarina_PlayCustomSequence(void) {
+    sRequestCustomSequence = true;
+    Audio_QueueSeqCmd(0x00000000 | ((u32)(SEQ_PLAYER_FANFARE) << 24) | ((u32)(0) << 0x10) |
+                      (u32)(NA_BGM_DUNGEON_APPEAR));
+}
+
+// Unused
+#ifdef NON_MATCHING
```

Comment:
Could you note what the issue is and/or include the decomp.me here if no one cracks it before merge?

Reply:
So is it agreed upon to actually commit decomp.me links to the repo? That seems strange to me but if it's only me, then I can add it

Reply:
I don't really care about commiting decompme urls, however I don't really see the point since the full context is already in the repo so can be worked on locally. However I do agree with adding a small note describing what isn't matching, to help when people look for Non-Matchings to fix.